src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentQueue.cs

```diff
@@ -627,6 +627,18 @@ public bool TryPeek(out T result)
             return false;
         }
 
+        /// <summary>
+        /// Removes all objects from the <see cref="ConcurrentQueue{T}"/>.
+        /// </summary>
+        public void Clear()
+        {
+            while (!IsEmpty)
+            {
+                Segment head = _head;
+                head.Clear();
```

Comment:
This is a lot of additional code.  Is it measurably more efficient than just doing:
```C#
T item;
while (TryDequeue(out item));
```
?  It looks like it's essentially a copy of the TryDequeue/TryRemove implementation, just with Clear/Clear naming, so I wouldn't expect it would be, unless I'm missing something.  We shouldn't be adding all of this extra complicated code unless it actually makes a meaningful difference.